src/hooks/baseHttp.ts

```diff
@@ -0,0 +1,387 @@
+import { SpansContainer, TracerGlobals } from '../globals';
+import {
+  getCurrentTransactionId,
+  getHttpInfo,
+  getHttpSpan,
+  getServiceData,
+  ServiceData,
+} from '../spans/awsSpan';
+import {
+  addHeaders,
+  getAWSEnvironment,
+  getEdgeHost,
+  getEventEntitySize,
+  getPatchedTraceId,
+  getRandomId,
+  isAwsService,
+  isEmptyString,
+  isKeepHeadersOn,
+  isTimeoutTimerEnabled,
+  isValidAlias,
+  lowerCaseObjectKeys,
+  safeExecute,
+  shouldPropagateW3C,
+} from '../utils';
+import { GlobalDurationTimer } from '../utils/globalDurationTimer';
+import * as logger from '../logger';
+import { BasicChildSpan } from '../types/spans/basicSpan';
+import { getW3CTracerPropagatorAdditionalHeaders } from '../utils/w3cUtils';
+import {
+  extractBodyFromEmitSocketEvent,
+  extractBodyFromWriteOrEndFunc,
+  httpDataToString,
+} from './httpUtils';
+import { URL } from 'url';
+import { parse as parseQuery } from 'querystring';
+import { shallowMask } from '../utils/payloadStringify';
+import { Agent } from './http';
+
+export const hostBlaclist = new Set(['127.0.0.1']);
+
+type HttpRequestTracingConfig = {
+  // Headers of the request, including user defined headers & added headers
+  headers: {};
+
+  // Headers added to the original headers
+  addedHeaders: {};
+
+  // The HTTP span that was created for this request (Will be updated in place in the request response lifecycle)
+  httpSpan?: BasicChildSpan;
+
+  requestData?: RequestData;
+
+  requestRandomId: string;
+
+  awsRequestId: string;
+  transactionId: string;
+};
+
+export type ParseHttpRequestOptions = {
+  agent?: Agent;
+  _defaultAgent?: Agent;
+  // eslint-disable-next-line no-undef
+  headers?: Record<string, string>;
+  method?: 'GET' | 'POST';
+  protocol?: string;
+  path?: string;
+  port?: number;
+  defaultPort?: number;
+  hostname?: string;
+  host?: string;
+  uri?: { hostname?: string };
+};
+
+export type RequestData = {
+  host?: string;
+  body?: any;
+  headers?: any;
+  path?: string;
+  truncated?: boolean;
+  port?: number;
+  uri?: string;
+  method?: string;
+  protocol?: string;
+  sendTime?: number;
+};
+
+export class BaseHttp {
```

Comment:
Since all these methods here are static, what's the value of having it inside a class? it can be broken into standalone functions exported from this module IMO